refactor: transition to flamework

homomorphist opened the pull request:
While certainly stupidly and utterly massive commit, I've managed to transition to Flamework will keeping nearly all features.

There are still a few things I want to finish up, but it's about on par with what it was previously.

homomorphist committed:
chore: install flamework + components

homomorphist committed:
refactor: transition to flamework

Agapurnis commented:
Before merging, I’d like to change a few more things.

- [ ] Add a GUI for the user balance
- [ ] Add a GUI for the ore count
- [ ] Introduce ore count limitations
- [ ] Refactor item placement controllers to make it easier to understand
- [ ] Convert remotes to take in arrays for bulk processing
  - Since ‘[@rbxts/netbuilder](https://github.com/Rimuy/netbuilder)’ doesn’t want to serialize things inside of arrays/serialize recursively, I’m going to just roll my own simple networking library to add support for serialization.

homomorphist committed:
feat: WIP custom net, bulk network operations

homomorphist committed:
chore: remove old debug code

homomorphist committed:
fix(net): client -> server invoke no longer takes Player

homomorphist committed:
refactor(net): specification is now a class, clean typings

- This removes the need for the useless property and also gets rid of an unused property.
- It also introducces return type validation, to be added later alongisde the parameter validation implementation.
- It also fixes a typing issue with the parameter validator types.
- It also gets rid of some useless assertions.
- Tidies up some documentation.
- Makes the class serialization format an interface.
- Remove incorrect remark about potential DOS attack.
- Better support circular serialization(?).
- Remove old file.
- This definitely should have been multiple commits.

homomorphist committed:
chore: remove old debug code

I swear I did this earlier? Not talking about the server one, either.
I thought I deleted this, then couldn't find the server one.
Then I just deleted the server, and now I didn't delete this one? Wack.

Agapurnis commented:
Y'know what, I'm practically using this as a new main branch. I might as well merge it.
